AppCenterDataStorage/AppCenterDataStorage/Internal/LocalStore/MSDBDocumentStore.m

```diff
@@ -37,18 +37,41 @@ - (instancetype)init {
     _expirationTimeColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSExpirationTimeColumnName]).unsignedIntegerValue;
     _downloadTimeColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSDownloadTimeColumnName]).unsignedIntegerValue;
     _operationTimeColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSOperationTimeColumnName]).unsignedIntegerValue;
-    _pendingOperationColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSPendingDownloadColumnName]).unsignedIntegerValue;
+    _pendingOperationColumnIndex = ((NSNumber *)columnIndexes[kMSAppDocumentTableName][kMSPendingOperationColumnName]).unsignedIntegerValue;
   }
   return self;
 }
 
 #pragma mark - Table Management
 
-// TODO work item created to track this implementation
-- (BOOL)createWithPartition:(NSString *)__unused partition
-                   document:(MSDocumentWrapper *)__unused document
-               writeOptions:(MSWriteOptions *)__unused writeOptions {
-  return YES;
+- (BOOL)upsertWithPartition:(NSString *)partition
+            documentWrapper:(MSDocumentWrapper *)documentWrapper
+                  operation:(NSString *_Nullable)operation
+                    options:(MSBaseOptions *)options {
+  NSDate *now = [NSDate date];
+  NSDate *expirationTime = [now dateByAddingTimeInterval:options.deviceTimeToLive];
+  NSString *insertQuery =
+      [NSString stringWithFormat:@"REPLACE INTO '%@' ('%@', '%@', '%@', '%@', '%@', '%@', '%@', '%@') "
+                                 @"VALUES ('%@', '%@', '%@', '%@', '%@', '%@', '%@', '%@')",
+                                 kMSAppDocumentTableName, kMSPartitionColumnName, kMSDocumentIdColumnName, kMSDocumentColumnName,
```

Comment:
This we would need to do a check on `partition` sent and decide which table we are working on right. It wont always be AppDocumentTable

Reply:
Actually on the AppDocumentTable, we would only be caching read actions right so if the action is on the app level, the Pending_operation is nil and the create, delete, update is populated only in the userDocumentTable .